### DISCUSSION TITLE
Event before body is downloaded, but after headers are fetched

### DISCUSSION THREAD

**Discussion by aPoCoMiLogin** [sindresorhus/got#2281] [Q&A] [answered]
Hi there, i'm looking for a way to inspect response headers before the response body is downloaded. Was looking around documentation, examples and the code, bu was unable to find anything. any help highly appreciated

**Comment by sindresorhus** [comment, marked as answer]
You can use the `response` event to access headers before the body is downloaded:

```javascript
import got from 'got';

const request = got('https://example.com');

request.on('response', (response) => {
	console.log('Headers:', response.headers);
	console.log('Status:', response.statusCode);
	console.log('Content-Type:', response.headers['content-type']);
	
	// You can even abort the download if needed
	if (response.headers['content-length'] > 1000000) {
		request.destroy();
	}
});

const response = await request;
```

Alternatively, if you're using streams, the headers are available as soon as the `response` event fires:

```javascript
const stream = got.stream('https://example.com');

stream.on('response', (response) => {
	console.log('Headers:', response.headers);
	// Body hasn't been downloaded yet
});

stream.pipe(someWriteStream);
```

The `response` event fires as soon as the headers are recieved, before any body data is downloaded, which is exactly what you need.
